### PR TITLE
Allow fixed homeserver

### PR DESCRIPTION
**Pull request by dklimpel**
Fixes: #114

Add parameter  `REACT_APP_SERVER` to fix homeserver at build-time.

### CODE REVIEW

File: src/components/LoginPage.js [hunk 1]
```diff
@@ -219,7 +222,8 @@ const LoginPage = ({ theme }) => {
             name="base_url"
             component={renderInput}
             label={translate("synapseadmin.auth.base_url")}
-            disabled={loading}
+            disabled={override_server ? true : loading}
```
Comment (awesome-manuel) [inline src/components/LoginPage.js:225]: IMHO this is better readable:
```suggestion
            disabled={override_server || loading}
```

File: src/components/LoginPage.js [hunk 2]
```diff
@@ -230,7 +234,7 @@ const LoginPage = ({ theme }) => {
 
   return (
     <Form
-      initialValues={{ base_url: base_url }}
+      initialValues={{ base_url: override_server ? override_server : base_url }}
```
Comment (awesome-manuel) [inline src/components/LoginPage.js:237]:
```suggestion
      initialValues={{ base_url: override_server || base_url }}
```
`||` returns the "truthy" value, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR

File: src/synapse/authProvider.js [hunk 3]
```diff
@@ -3,6 +3,11 @@ import { fetchUtils } from "react-admin";
 const authProvider = {
   // called when the user attempts to log in
   login: ({ base_url, username, password }) => {
+    // force homeserver for protection in case the form is manipulated
+    base_url = process.env.REACT_APP_SERVER
+      ? process.env.REACT_APP_SERVER
+      : base_url;
```
Comment (awesome-manuel) [inline src/synapse/authProvider.js:9]:
```suggestion
    base_url = process.env.REACT_APP_SERVER || base_url;
```
See other suggestions.

File: src/components/LoginPage.js [hunk 4]
```diff
@@ -82,6 +82,7 @@ const LoginPage = ({ theme }) => {
   const setLocale = useSetLocale();
   const translate = useTranslate();
   const base_url = localStorage.getItem("base_url");
+  const override_server = process.env.REACT_APP_SERVER;
```
Comment (awesome-manuel) [inline src/components/LoginPage.js:85]: Maybe `cfg_base_url` instead of `override_server`.

**Review by awesome-manuel — CHANGES_REQUESTED** [review]
Some minor changes requested.

**Review by awesome-manuel — APPROVED** [review]

**Comment by rainer-geiger** [comment]
Can you include something like this in your dockerfile to pass 'REACT_APP_SERVER' as arg from docker-compose.yml?

```
ARG REACT_APP_SERVER
ENV REACT_APP_SERVER $REACT_APP_SERVER
```

If I'm not completely wrong, then you should be able to user fixed homeserver from docker-compose like this:

```
build:
      [...]
      args:
        - REACT_APP_SERVER=https://matrix.yourdomain.com
```

**Pull request merged**